src/stepper.c

```diff
@@ -209,39 +212,35 @@ void
 command_queue_step(uint32_t *args)
 {
     struct stepper *s = stepper_oid_lookup(args[0]);
-    struct stepper_move *m = move_alloc();
-    m->interval = args[1];
-    m->count = args[2];
-    if (!m->count)
+    struct stepper_move *new_event = mq_alloc_event();
+    new_event->interval = args[1];
+    new_event->count = args[2];
+    if (!new_event->count)
         shutdown("Invalid count parameter");
-    m->add = args[3];
-    m->next = NULL;
-    m->flags = 0;
+    new_event->add = args[3];
+    new_event->flags = 0;
 
     irq_disable();
     uint8_t flags = s->flags;
     if (!!(flags & SF_LAST_DIR) != !!(flags & SF_NEXT_DIR)) {
         flags ^= SF_LAST_DIR;
-        m->flags |= MF_DIR;
+        new_event->flags |= MF_DIR;
     }
     flags &= ~SF_NO_NEXT_CHECK;
-    if (m->count == 1 && (m->flags || flags & SF_LAST_RESET))
+    if (new_event->count == 1 && (new_event->flags || flags & SF_LAST_RESET))
         // count=1 moves after a reset or dir change can have small intervals
         flags |= SF_NO_NEXT_CHECK;
     flags &= ~SF_LAST_RESET;
-    if (s->count) {
-        s->flags = flags;
-        if (s->first)
-            *s->plast = m;
-        else
-            s->first = m;
-        s->plast = &m->next;
-    } else if (flags & SF_NEED_RESET) {
-        move_free(m);
-    } else {
-        s->flags = flags;
-        s->first = m;
```

Comment:
This overwrites s->first without setting `s->plast = &m->next`

Reply:
FYI, the next line of code (`stepper_load_next()`) immediately pops `s->first` from the list and the list goes back to being empty.  So, I don't think there is a defect in the current code.

This isn't a "fast path" though, so there's no harm in using different list management here if it simplifies the code.